### ISSUE TITLE
Failed to run npm install

### ISSUE THREAD

**Issue by fbonawiede** [zodern/mup-aws-beanstalk#99]
METEOR@1.5.4.2

```
=> Setting up
=> Ensuring IAM Roles and Instance Profiles are setup
Building App Bundle Locally
=> Archiving Bundle
  10% Archived
  20% Archived
  30% Archived
  40% Archived
  50% Archived
  60% Archived
  70% Archived
  80% Archived
  90% Archived
  100% Archived
=> Uploading bundle
  Uploaded 31%
  Uploaded 62%
  Uploaded 94%
  Uploaded 100%
  Finishing upload. This could take a couple minutes
=> Creating Version
=> Configuring Beanstalk Environment
=> Deploying new version
=> Waiting for Beanstalk Environment to finish updating
  Env Event: Rolling with Additional Batch deployment policy enabled. Launching  a new batch of 1 additional instance(s).
  Env Event: Batch 1: 1 EC2 instance(s) [i-01c459b7f22d8015c] launched. Deploying application version '2'.
  Env Event: Added instance [i-01c459b7f22d8015c] to your environment.
  Env Event: During an aborted deployment, some instances may have deployed the new application version. To ensure all instances are running the same version, re-deploy the appropriate application version.
  Env Event: Failed to deploy application.
  Env Event: Unsuccessful command execution on instance id(s) 'i-01c459b7f22d8015c'. Aborting the operation.
  Env Event: Command execution completed on all instances. Summary: [Successful: 0, Failed: 1].
  Env Event: [Instance: i-01c459b7f22d8015c] Command failed on instance. Return code: 1 Output: (TRUNCATED)...opt/elasticbeanstalk/containerfiles/ebnode.py", line 180, in npm_install
    raise e
subprocess.CalledProcessError: Command '['/opt/elasticbeanstalk/node-install/node-v12.16.2-linux-x64/bin/npm', '--production', 'install']' returned non-zero exit status 1.
Hook /opt/elasticbeanstalk/hooks/appdeploy/pre/50npm.sh failed. For more detail, check /var/log/eb-activity.log using console or EB CLI.
  Env Event: Failed to run npm install. Snapshot logs for more details.
  Env Event: Retrieving logs prior to instance(s) termination. Logs will be available for an hour in the environment management console and at elasticbeanstalk-eu-north-1-731427829640/resources/environments/logs/bundle/e-pdj2bynh2c.
  Env Event: Terminating excess instance(s): [i-01c459b7f22d8015c].
  Env Event: Command execution completed on all instances successfully.
  Env Event: [Instance: i-01c459b7f22d8015c] Successfully finished bundling 16 log(s)
  Env Event: Excess instance(s) terminated.
App is running at mup-env-mtr-rusning.eba-4iiytguc.eu-north-1.elasticbeanstalk.com
=> Finding old versions
=> Removing old versions
=> Checking Certificate Status
Certificate has been issued
=> Updating Beanstalk SSL Config
```

It seems like it fails here:
```
###############                                                           21.5%
##################################                                        47.4%
#####################################################                     74.1%
######################################################################## 100.0%
  Computing checksum with sha256sum
  Checksums matched!
  Now using node v4.9.0 (npm v2.15.11)
  Creating default alias: [0;32mdefault[0m [0;90m->[0m [0;32m4.9.0[0m ([0;90m->[0m [0;32mv4.9.0[0m)
  Now using node v4.9.0 (npm v2.15.11)
  [0;32mdefault[0m [0;90m->[0m [0;32m4.9.0[0m ([0;90m->[0m [0;32mv4.9.0[0m)
  /.nvm/versions/node/v4.9.0/bin/npm -> /.nvm/versions/node/v4.9.0/lib/node_modules/npm/bin/npm-cli.js
  npm@4.6.1 /.nvm/versions/node/v4.9.0/lib/node_modules/npm
[2020-05-06T08:11:37.006Z] INFO  [2554]  - [Application deployment 2@3/StartupStage0/AppDeployPreHook/50npm.sh] : Starting activity...
[2020-05-06T08:11:47.403Z] INFO  [2554]  - [Application deployment 2@3/StartupStage0/AppDeployPreHook/50npm.sh] : Activity execution failed, because: + /opt/elasticbeanstalk/containerfiles/ebnode.py --action npm-install
  
  > mup-mtr-rusning@2.0.0 preinstall /tmp/deployment/application
  > cd programs/server && npm install
  
  npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but npm-shrinkwrap.json was generated for lockfileVersion@0. I'll try to do my best with it!
  npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
  
  > fibers@1.0.15 install /tmp/deployment/application/programs/server/node_modules/fibers
  > node build.js || nodejs build.js
  
  make: Entering directory `/tmp/deployment/application/programs/server/node_modules/fibers/build'
    CXX(target) Release/obj.target/fibers/src/fibers.o
  ../src/fibers.cc:70:58: error: ‘Handle’ has not been declared
    void Reset(Isolate* isolate, Persistent<T>& persistent, Handle<T> handle) {
```

**Comment by joaocarloscabral** [comment]
Same situation here.

@fbonawiede Have you found any solution yet?

**Comment by zodern** [comment]
Could you both please answer a few question?

1. What Meteor version are you using?
2. Did deploying work before?
3. Did you update meteor or any packages since then?

**Comment by joaocarloscabral** [comment]
I was trying to deploy in a new server.
I solved my problem creating a swapfile with .ebextensions. (#72)

Thanks @zodern

**Comment by fbonawiede** [comment]
> Could you both please answer a few question?
> 
> 1. What Meteor version are you using?
> 2. Did deploying work before?
> 3. Did you update meteor or any packages since then?

@zodern 
1. METEOR@1.5.4.2
2. I've not been able to successfully deploy it.

@joaocarloscabral I've been struggling with .ebextensions but not got it to work. I added the following file in .ebextentions folder in the project root as 01_fix_permissions.config

```
files:
  "/opt/elasticbeanstalk/hooks/appdeploy/pre/49_change_permissions.sh":
    mode: "000755"
    owner: root
    group: root
    content: |
      #!/usr/bin/env bash
      sudo chown -R ec2-user:ec2-user tmp/
```

**Comment by joaocarloscabral** [comment]
@fbonawiede .ebextenstions is not avaiable yet.
Check the #72 pull request.

You could use a @settlin/mup-aws-beanstalk fork until the oficial release is not avaible to use .ebxtensions files.

**Comment by fbonawiede** [comment]
@joaocarloscabral can you provide me with the config file that creates the swapfile?

I am using the fork and the config files now ends up in the deployed zip

**Comment by joaocarloscabral** [comment]
The problem I was experiencing was out of memory, so it made sense for me to create the swapfile.

If your problem is the same, the swapfile can help.

Here is:
.ebextensions/setup_swap.sh
```
#!/bin/bash

SWAPFILE=/var/swapfile
SWAP_MEGABYTES=2048

if [ -f $SWAPFILE ]; then
	echo "Swapfile $SWAPFILE found, assuming already setup"
	exit;
fi

/bin/dd if=/dev/zero of=$SWAPFILE bs=1M count=$SWAP_MEGABYTES
/bin/chmod 600 $SWAPFILE
/sbin/mkswap $SWAPFILE
/sbin/swapon $SWAPFILE
```

.extensions/swap.config
```
container_commands:
    01setup_swap:
        command: "bash .ebextensions/setup_swap.sh"
``` 


Maybe you can use a unique file, like this:
.ebextensions/swap.config
```
commands:
    enable_swap_memory:
        command: "fallocate -l 1G /swapfile && chmod 600 /swapfile && mkswap /swapfile && swapon /swapfile"
        ignoreErrors: true

```

**Comment by applew** [comment]
@fbonawiede I was getting the same error "Hook /opt/elasticbeanstalk/hooks/appdeploy/pre/50npm.sh failed.". Turns out I did not set the correct node version for my Meteor app.

Thought I would leave this here in case someone else has the same issue.

Dropping this in my mup.js resolved the problem:
    customBeanstalkConfig: [
      {
        namespace: 'aws:elasticbeanstalk:container:nodejs',
        option: 'NodeVersion',
        value: '8.15.1'
      }
    ]

**Comment by zodern** [comment]
The issues caused by using the wrong node version will be fixed in 0.7 (#142).
